homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+        }
+
+        return self.async_show_form(
+            step_id="user",
+            description_placeholders=self.device_config,
+            data_schema=vol.Schema(data),
+            errors=errors,
+        )
+
+    async def _create_entry(self):
+        """Create entry for device."""
+
+        return self.async_create_entry(
+            title=self.device_config[CONF_NAME], data=self.device_config
+        )
+
+
+class AmcrestOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Amcrest device options."""
+
+    def __init__(self, config_entry):
+        """Initialize Amcrest device options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.device = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Amcrest device options."""
+        self.device = self.hass.data[DOMAIN][DEVICES][self.config_entry.data[CONF_NAME]]
+        return await self.async_step_configure_stream()
+
+    async def async_step_configure_stream(self, user_input=None):
+        """Manage the Amcrest device options."""
+        if user_input is not None:
+            self.options.update(user_input)
+            return self.async_create_entry(title="", data=self.options)
+
+        return self.async_show_form(
+            step_id="configure_stream",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_STREAM_SOURCE,
+                        default=self.config_entry.options.get(
+                            CONF_STREAM_SOURCE, DEFAULT_STREAM_SOURCE
+                        ),
+                    ): vol.In(STREAM_SOURCE_LIST),
+                    vol.Optional(
+                        CONF_FFMPEG_ARGUMENTS,
+                        default=self.config_entry.options.get(
+                            CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+                        ),
+                    ): str,
+                    vol.Optional(
+                        CONF_RESOLUTION,
+                        default=self.config_entry.options.get(
+                            CONF_RESOLUTION, DEFAULT_RESOLUTION
+                        ),
+                    ): vol.All(vol.In(RESOLUTION_LIST)),
+                    vol.Optional(
+                        CONF_AUTHENTICATION,
+                        default=self.config_entry.options.get(
+                            CONF_AUTHENTICATION, DEFAULT_AUTHENTICATION
+                        ),
+                    ): vol.All(vol.In(AUTHENTICATION_LIST)),
+                    vol.Optional(
+                        CONF_CONTROL_LIGHT,
+                        default=self.config_entry.options.get(
+                            CONF_CONTROL_LIGHT, DEFAULT_CONTROL_LIGHT
+                        ),
+                    ): bool,
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
```

Comment:
Why would we make the scan interval configurable ? We should pick the most optimal one that the device can handle.

Reply:
certainly, if all devices behaved the same. But unfortunately in this case it is not true

Reply:
It's my understanding that scan interval is only used if the optional POLLING method is used for either audio or motion detection binary_sensors.   I'm not aware that it has any other purpose.   Certainly the new event subscription (aka streaming) method for these sensors that Phil added a few months ago does not require it. 

The problem Phil found was that not all Amcrest cameras supported subscription to camera events.  (They must have been really old cameras).  Consequently for backward compatibility, he added two new binary sensors "audio_detected_polled" and "motion_detected_polled" to be alternatives.

Personally I would not have done it this way, especially now that blade is adding new sensors and we end up with two of every sensor.    I suggest adding a new config flow option "**Event_Method**"   (default "stream" - option "polled").    Then a sub option to the polled method is the scan interval.